packages/cdktf/lib/tfExpression.ts

```diff
@@ -171,6 +172,11 @@ function markAsInner(arg: any) {
     return;
   }
 
+  if (arg instanceof Construct) {
```

Comment:
I wonder if this fixes the infinite loop problem we encountered in some of the converted code snippets